src/Polyfill/IndexRange/RuntimeHelpers.cs

```diff
@@ -0,0 +1,47 @@
+// <auto-generated />
+
+#if (NET46X && FeatureValueTuple) || NET47X || NET48X || NETSTANDARD2_0 || NETCOREAPP2X
+
+namespace System.Runtime.CompilerServices;
+
+using Link = System.ComponentModel.DescriptionAttribute;
+
+/// <summary>
+/// Provides a set of static methods and properties that provide support for compilers.
+/// This class cannot be inherited.
+/// </summary>
+#if PolyPublic
+public
+#endif
+static partial class RuntimeHelpers
+{
+    /// <summary>
+    /// Slices the specified array using the specified range.
+    /// </summary>
+    /// <typeparam name="T">The type of the elements in the array.</typeparam>
+    /// <param name="array">The array to slice.</param>
+    /// <param name="range">An object that determines the portion of array to include in the slice.</param>
+    /// <returns>The subarray defined by range.</returns>
+    [Link("https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimehelpers.getsubarray")]
+    public static T[] GetSubArray<T>(T[] array, Range range)
+    {
+        if (array == null)
+        {
+            throw new ArgumentNullException(nameof(array));
+        }
+
+        var (offset, length) = range.GetOffsetAndLength(array.Length);
+
+        if (length == 0)
+        {
+            return [];
+        }
+
+        var dest = new T[length];
+
+        Array.Copy(array, offset, dest, 0, length);
+
+        return dest;
```

Comment:
Do you prefer this implementation or one that will be released with .NET 9?
https://github.com/dotnet/runtime/blob/v9.0.0-preview.3.24172.9/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs#L47-L54